### PR TITLE
Refactor of Icelandic Locale For Dehumanize Support

### PR DESCRIPTION
**Pull request by anishnya**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [ ] 🧪  Added **tests** for changed code.
- [ ] 🛠️  All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [ ] 🧹  All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [ ] 📚  Updated **documentation** for changed code.
- [ ] ⏩  Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes

Refactored Icelandic Locale for Future Dehumanize Support (#961 and #970).

### CODE REVIEW

File: arrow/locales.py [hunk 1]
```diff
@@ -2755,11 +2755,13 @@ class MoroccoArabicLocale(ArabicLocale):
...
-        elif delta > 0:
-            form = form[1]
-            # FIXME: handle when delta is 0
+
+        if isinstance(form, Mapping):
+            if delta < 0:
+                form = form["past"]
+            elif delta >= 0:
+                form = form["future"]
+                # FIXME: handle when delta is 0
```
Comment (systemcatch) [inline arrow/locales.py:2764]: I guess this is now fixed when `delta=0`?

Comment (anishnya) [reply arrow/locales.py]: So I think there are two options here. We default to the future value when the case is that the delta for a particular unit is zero, or we implicitly error out like we currently do. In an ideal scenario, I would add a raise value error here, but I fear that would probably break the use of the locale, given how common the use of a delta of zero is.

Comment (systemcatch) [reply arrow/locales.py]: I would tend to think error out is better as it is probably less surprising.

Comment (systemcatch) [reply arrow/locales.py]:
```shell
(arrow) chris@Z490:~/arrow$ python
Python 3.9.1 (default, Feb 17 2021, 16:56:42) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> start=arrow.utcnow()
>>> start.humanize(other=start, locale="is")
'rétt í þessu'
>>> start.humanize(other=start, granularity="hour",locale="is")
'eftir 0 tíma'
```

**Review by systemcatch — APPROVED** [review]

**Review by krisfremen — APPROVED** [review]
LGTM

**Pull request merged**